warehouse/query-core/src/main/java/datawave/query/iterator/logic/TermFrequencyExcerptIterator.java

```diff
@@ -315,28 +386,244 @@ public void next() throws IOException {
+                    trimEnd(termsToOutput, afterIndex, start);
+                }
+            }
+            // join everything together with spaces while skipping null offsets
+            return joiner.join(termsToOutput);
+        }
+    }
+
+    /**
+     * Trim down both side of the excerpt to the size that we want
+     *
+     * @param termsToOutput
+     *            the terms to create a phrase from
+     * @return the trimmed array
+     */
+    private String[] bothTrim(String[] termsToOutput) {
+        int expandedMid = (endOffset - startOffset) / 2;
+        int start = (int) (expandedMid - origHalfSize);
+        trimBeginning(termsToOutput, expandedMid, start);
+        start = (int) (expandedMid + origHalfSize);
+        trimEnd(termsToOutput, expandedMid, start);
+        return termsToOutput;
+    }
+
+    /**
+     * Trims off the front to get the correct size
+     *
+     * @param termsToOutput
+     *            the terms to create a phrase from
+     * @param beforeIndex
+     *            the index in the array to start at
+     * @param start
+     *            the index in the array where we want to start setting values to null at
+     */
+    private void trimBeginning(String[] termsToOutput, int beforeIndex, int start) {
+        boolean startNull = false;
+        for (int k = beforeIndex; k >= 0; k--) {
+            if (!startNull && k < start) {
+                startNull = true;
+            }
+            if (startNull) {
+                termsToOutput[k] = null;
+            } else {
+                if (termsToOutput[k] == null) {
+                    start--;
+                }
+            }
+        }
+    }
+
+    /**
+     * Trims off the end to get the correct size
+     *
+     * @param termsToOutput
+     *            the terms to create a phrase from
+     * @param afterIndex
+     *            the index in the array to start at
+     * @param start
+     *            the index in the array where we want to start setting values to null at
+     */
+    private void trimEnd(String[] termsToOutput, int afterIndex, int start) {
+        boolean startNull = false;
+        for (int k = afterIndex; k < termsToOutput.length; k++) {
+            if (!startNull && k > start) {
+                startNull = true;
+            }
+            if (startNull) {
+                termsToOutput[k] = null;
+            } else {
+                if (termsToOutput[k] == null) {
+                    start++;
+                }
+            }
+        }
+    }
+
+    /**
+     * Looks for hit phrases (not separate hit terms) and puts the whole phrase in brackets
+     *
+     * @param terms
+     *            the terms to create a phrase from
+     */
+    private void checkForHitPhrase(WordsAndScores[] terms) {
+        ArrayList<String> hitPhrases = new ArrayList<>();
+        // checks for phrases (anything in the hit list with a space in it) and adds them to a new arrayList
+        for (String s : hitTermsList) {
```

Comment:
There's nothing preventing a value from having a space in it. This code will work but might discover more phrases than actually exist. Longer term we should have a separate attribute type for hit term phrases.

Reply:
I agree. Right now this code will never actually be triggered when running because I learned that datawave does not pass down the whole phrases into the hit excerpt code currently. plans for part 2 of this include finding a way to pass that down and differentiate the original phrases from the broken up pieces.